pkg/gofr/datasource/solr/solr.go

```diff
@@ -0,0 +1,215 @@
+func (c *Client) UseLogger(logger interface{}) {
+	if l, ok := logger.(Logger); ok {
+		c.logger = l
+	}
+}
+
+// UseMetrics sets the metrics for the Cassandra client which asserts the Metrics interface.
+func (c *Client) UseMetrics(metrics interface{}) {
+	if m, ok := metrics.(Metrics); ok {
+		c.metrics = m
+	}
+}
+
+// Connect establishes a connection to Solr and registers metrics using the provided configuration when the client was Created.
+func (c *Client) Connect() {
+	c.logger.Infof("connecting to Solr at %v", c.url)
+
+	solrBuckets := []float64{.05, .075, .1, .125, .15, .2, .3, .5, .75, 1, 2, 3, 4, 5, 7.5, 10}
+	c.metrics.NewHistogram("app_solr_stats", "Response time of Solr operations in milliseconds.", solrBuckets...)
+
+	return
+}
+
+func (c *Client) HealthCheck(ctx context.Context) (any, error) {
+	return nil, nil
+}
+
+// Search searches documents in the given collections based on the parameters specified.
+// This can be used for making any queries to SOLR
+func (c *Client) Search(ctx context.Context, collection string, params map[string]interface{}) (interface{}, error) {
+	url := c.url + collection + "/select"
+
+	defer c.sendOperationStats(&QueryLog{Type: "Search", Url: url}, time.Now())
+
+	return call(ctx, "GET", url, params, nil)
+}
+
+// Create makes documents in the specified collection. params can be used to send parameters like commit=true
+func (c *Client) Create(ctx context.Context, collection string, document *bytes.Buffer,
+	params map[string]interface{}) (interface{}, error) {
+	url := c.url + collection + "/update"
+
+	defer c.sendOperationStats(&QueryLog{Type: "Create", Url: url}, time.Now())
+
+	return call(ctx, "POST", url, params, document)
+}
+
+// Update updates documents in the specified collection. params can be used to send parameters like commit=true
+func (c *Client) Update(ctx context.Context, collection string, document *bytes.Buffer,
+	params map[string]interface{}) (interface{}, error) {
+	url := c.url + collection + "/update"
+
+	defer c.sendOperationStats(&QueryLog{Type: "Update", Url: url}, time.Now())
+
+	return call(ctx, "POST", url, params, document)
+}
+
+// Delete deletes documents in the specified collection. params can be used to send parameters like commit=true
+func (c *Client) Delete(ctx context.Context, collection string, document *bytes.Buffer,
+	params map[string]interface{}) (interface{}, error) {
+	url := c.url + collection + "/update"
+
+	defer c.sendOperationStats(&QueryLog{Type: "Delete", Url: url}, time.Now())
+
+	return call(ctx, "POST", url, params, document)
+}
+
+// ListFields retrieves all the fields in the schema for the specified collection.
+// params can be used to send query parameters like wt, fl, includeDynamic etc.
+func (c *Client) ListFields(ctx context.Context, collection string, params map[string]interface{}) (interface{}, error) {
+	url := c.url + collection + "/schema/fields"
+
+	defer c.sendOperationStats(&QueryLog{Type: "ListField", Url: url}, time.Now())
```

Comment:
Shouldn't it match the method name?

```suggestion
	defer c.sendOperationStats(&QueryLog{Type: "ListFields", Url: url}, time.Now())
```